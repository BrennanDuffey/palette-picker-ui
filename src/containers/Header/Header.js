import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

function Header(props) {
  return (
    <header>
    <h1>Palette Picker</h1>
      {!props.user.name && <NavLink to='/home/register'>LOGIN</NavLink>}
      {props.user.name && <NavLink to='/home/'>SIGN OUT</NavLink>}
    </header>
  );
};

export const mapStateToProps = (state) => ({
  user: state.user
});

export default connect(mapStateToProps)(Header);