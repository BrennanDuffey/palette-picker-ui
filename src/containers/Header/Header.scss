.header {
  display: flex;
  justify-content: space-between;
  padding: 20px;
};

.login-navlink {
  margin: 10px;
}