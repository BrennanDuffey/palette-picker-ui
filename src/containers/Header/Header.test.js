import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header', () => {
  let wrapper;
  let mockUser;

  beforeEach(() => {
    mockUser = {
      id: 2,
      username: 'something'
    };
    wrapper = shallow(
      <Header user={mockUser} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot for no username', () => {
    mockUser = {};
    wrapper = shallow(
      <Header user={mockUser} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});