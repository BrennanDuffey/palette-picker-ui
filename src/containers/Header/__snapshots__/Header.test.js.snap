// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  className="header"
>
  <NavLink
    className="main-header"
    to="/newpalette"
  >
    <h1>
      Palette Picker
    </h1>
  </NavLink>
  <div
    className="user-nav"
  >
    <NavLink
      to="/projects"
    >
      <p>
        YOUR PROJECTS
      </p>
    </NavLink>
    <NavLink
      to="/newpalette"
    >
      <p>
        NEW PALETTE
      </p>
    </NavLink>
    <a>
      <p>
        SIGN OUT
      </p>
    </a>
  </div>
</header>
`;

exports[`Header should match snapshot for no username 1`] = `
<header
  className="header"
>
  <NavLink
    className="main-header"
    to="/newpalette"
  >
    <h1>
      Palette Picker
    </h1>
  </NavLink>
  <NavLink
    to="/login"
  >
    <p
      className="login-navlink"
    >
      LOGIN
    </p>
  </NavLink>
</header>
`;
